ha_vpn_over_interconnect_5GB_attach/main.tf

```diff
@@ -0,0 +1,295 @@
+# HA VPN over Cloud Interconnect 5GB VLAN attachments (Dedicated Interconnect)
+
+# [START cloudinterconnect_ha_vpn_over_interconnect_5gb_attachments]
+
+# Create all resources in the same region, which you can specify at the provider-level
+provider "google" {
+  project = "your-project"
+  region = "us-east4" # Region must support creation of all new attachments on Dataplane v2
+}
+
+# Network
+resource "google_compute_network" "network_havpn_ic" {
+  name = "network-havpn-ic"
+  auto_create_subnetworks = false
+  routing_mode = "GLOBAL"
+}
+
+# Subnet
+resource "google_compute_subnetwork" "subnet_havpn_ic" {
+  name = "subnet-havpn-ic"
+  ip_cidr_range = "192.168.1.0/24"
+  network = google_compute_network.network_havpn_ic.self_link
+}
+
+# Begin Cloud Interconnect tier
+# Create Interconnect Cloud Router
+
+resource "google_compute_router" "ic_router" {
+  name = "ic-router"
+  network = google_compute_network.network_havpn_ic.self_link
+  encrypted_interconnect_router = true
+  bgp {
+    asn = 65000
+  }
+}
+
+# Optional: Reserve regional internal IP ranges for the HA VPN gateway interfaces
+# on each VLAN attachment
+
+resource "google_compute_address" "address_vpn_ia_1" {
+  name          = "address-vpn-ia-1"
+  address_type  = "INTERNAL"
+  purpose       = "IPSEC_INTERCONNECT"
+  address       = "192.168.20.0"
+  prefix_length = 29 # Allows you to reserve up to 8 IP addresses
+  network       = google_compute_network.network_havpn_ic.self_link
+}
+
+resource "google_compute_address" "address_vpn_ia_2" {
+  name          = "address-vpn-ia-2"
+  address_type  = "INTERNAL"
+  purpose       = "IPSEC_INTERCONNECT"
+  address       = "192.168.21.0"
+  prefix_length = 29 # Allows you to reserve up to 8 IP addresses
+  network       = google_compute_network.network_havpn_ic.self_link
+}
+
+# Create encrypted VLAN attachments
+
+resource "google_compute_interconnect_attachment" "ia_1" {
+  name = "ia-1"
+  router = google_compute_router.ic_router.self_link
+  interconnect = "https://www.googleapis.com/compute/v1/projects/your-project/global/interconnects/interconnect-zone1"
```

Comment:
Do you actually need to be pointing to another project, or is using your same project an option?
If both ways are valid, I suggest showing the current project via https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/project and adding a comment saying that you can point to a different project for the interconnect.

Reply:
Both ways are valid, so that's a good suggestion

Reply:
Done